src/omero_cli_render.py

```diff
@@ -636,11 +637,21 @@ def set(self, args):
                 cindices.append(i)
             rangelist.append([c.start, c.end])
             colourlist.append(c.color)
+        return (namedict, cindices, rangelist, colourlist)
+
+    @gateway_required
+    def set(self, args):
+        """ Implements the 'set' command """
+        data = self._load_rendering_settings(
+            args.channels, session=self.client.getSession())
 
         iids = []
         for img in self.render_images(self.gateway, args.object, batch=1):
             iids.append(img.id)
 
+            # Extract settings from dictionary
+            (namedict, cindices, rangelist, colourlist) = self._read_channels(
+                data)
```

Comment:
This could be moved outside the for loop, line 672 `greyscale = ...` too.

Reply:
:+1: I had moved this block within the for loop as in my in-progress branch adding support for `n` images/`n` rendering files, the settings are parsed for each image. In the default scenario, I agree this is unnecessary.  b033bba should migrate the block back outside the `for` loop

Reply:
Ah of course, in these cases it makes sense. Sorry, feel free to revert the commit and leave it within the the for loop.